network-api/networkapi/wagtailpages/pagemodels/blog/mock_search_blog_index.py

```diff
@@ -0,0 +1,29 @@
+from wagtail.core.models import Page
+from wagtail.contrib.routable_page.models import RoutablePageMixin, route
+from networkapi.wagtailpages.utils import get_locale_from_request, get_default_locale
+from .blog import BlogPage
+
+
+class MockSearchBlogIndex(RoutablePageMixin, Page):
+
+    template = 'wagtailpages/mock_search_blog_index.html'
+
+    def search_pages(self, locale, search_query):
+        return BlogPage.objects.live().public().filter(locale=locale).search(search_query)
+
+    @route(r"^search/$")
+    def blog_page_search(self, request, *args, **kwargs):
```

Comment:
**Suggestion:** I am not sure we need the `RoutablePageMixin`. We could have the same behaviour on the blog index page directly. This is assuming that the blog index displays all child pages by default and the "search" acts as a type of filter. In that case, we could move the exact functionality to [`get_context`](https://docs.wagtail.org/en/stable/reference/pages/model_reference.html#wagtail.core.models.Page.get_context) method.

Reply:
Actually, I does not even need the assumption that the "unsearched" index shows all pages. That's just how it's designed in the research hub.

Reply:
it's nice to have a dedicated URL with the word `search` in it, though

Reply:
Could call the query parameter search 😉